src/com/vogon101/Graphics/Controls/ButtonControl.scala

```diff
@@ -0,0 +1,121 @@
+package com.vogon101.Graphics.Controls
+
+import com.colofabrix.scala.geometry.shapes.{Circle, Box}
+import com.colofabrix.scala.math.Vector2D
+import com.colofabrix.scala.tankwar.integration.TankEvaluator
+import org.lwjgl.opengl.GL11
+import org.newdawn.slick.opengl.Texture
+
+/**
+ * Created by Freddie on 18/05/2015.
+ *
+ * This class contains a Button. It has a render method, so is responsable for its own rendering
+ * as well as a Unit that it is it's OnClick action. Requires a Box as a renderable/clickable area and
+ * a Unit as an action
+ *
+ */
+class ButtonControl (_screenArea: Box, action : (Int) => Unit) {
+
+  //TODO: Make timing in seconds not steps
+  // ->TODO: Add a universal DeltaTime for the project
+
+  require(_screenArea != null, "Must have a screen area")
+  require(action != null, "Must have an action")
+
+
+  /**
+   * The minimum amount of time between clicks
+   */
+  val MIN_CLICK_TIMER = 25;
+  private var clickTimer: Int = MIN_CLICK_TIMER;
+
+  /**
+   * Increment the click timer by one
+   */
+  def clickTimerInc() {clickTimer += 1}
+
+  private var _texture: Texture = null;
+
+  /**
+   * @return The Box of the screen area
+   */
+  def area = _screenArea
+
+  /**
+   * @return The Texture of the box - UNIMPLEMENTED
+   */
+  def texture = _texture
+
+  /**
+   * Set the texture
+   * @param value The Texture to set it to - UNIMPLEMENTED
+   */
+  def texture_= (value: Texture): Unit = _texture = value
+
+
+
+  /**
```

Comment:
Redundant, check com.colofabrix.scala.geometry.shapes.Box.overlaps(...) and more generally com.colofabrix.scala.geometry.shapes.abstracts.Shape.